examples/inline-yaml.cf

```diff
@@ -0,0 +1,49 @@
+#+begin_src cfengine3
+bundle agent example_inline_yaml
+#@ brief Example illustrating inline yaml
+{
+  vars:
+      # YAML requires "---" header (followed by newline)
+      # NOTE \n is not interpreted as a newline, instead use $(const.n)
+
+      "yaml_multi_line" data => '---
+- key1: value1
+- key2: value2';
+
+
+      "yaml_single_line" data => '---$(const.n)- key1: value1$(const.n)- key2: value2';
+
+  reports:
+      "Data container defined from yaml_multi_line: $(with)"
+        with => string_mustache( "{{%-top-}}", @(yaml_multi_line) );
```

Comment:
I think the example might be clearer by using `with => storejson(@(yaml_multi_line));` I find the mustache stuff a bit noisy and confusing (and just a touch longer to type ;).

Reply:
My favorite is `with => format("%S", variable)` because it works for any variable :)

Reply:
@tzz not bad actually! That sort of improves the output in a way.

Gives me
```
R: yml [{"key1":"value1"},{"key2":"value2"}]
```

(on one line)

instead of the multi-line and structured output of `string_mustache` or `storejson`.

Feature request would be supporting this though right? ;)

```
  "Data container defined from yaml_multi_line: $(yaml_multi_line)";
```

That would be the "easiest" and most obvious I think. Kind of like python `f"` string formatting with variable references inside. :)

Reply:
I think that multiline output for data is much easier to read, so I tend to avoid format() unless I am on some older version because as far as I know, it won't do multiline output.

I tend to use mustache because then if I want to render parts of the data it's easier and already there, but also I didn't realize that storejson() would produce multiline output. I can switch to that unless @tzz knows magic format() to make it multiline.


I prefert

Reply:
> Feature request would be supporting this though right? ;)
> 
> ```
>   "Data container defined from yaml_multi_line: $(yaml_multi_line)";
> ```
> 
> That would be the "easiest" and most obvious I think. Kind of like python `f"` string formatting with variable references inside. :)

Absolutely. I proposed something like that years ago and it was rejected by the team, so we ended up converging (ha!) on the `with` attribute. Maybe the current team is willing to reconsider.

Reply:
> unless @tzz knows magic format() to make it multiline.

Nope, sorry... Craig's way is best.

Reply:
> > Feature request would be supporting this though right? ;)
> > ```
> >   "Data container defined from yaml_multi_line: $(yaml_multi_line)";
> > ```
> > 
> > 
> > That would be the "easiest" and most obvious I think. Kind of like python `f"` string formatting with variable references inside. :)
> 
> Absolutely. I proposed something like that years ago and it was rejected by the team, so we ended up converging (ha!) on the `with` attribute. Maybe the current team is willing to reconsider.

https://tracker.mender.io/browse/CFE-3552